src/buildConfig.js

```diff
@@ -44,7 +46,7 @@ export default function (options) {
   const configBasedir = options.configBasedir || false
 
   if (rawConfig) {
-    const configDir = configBasedir || process.cwd()
+    const configDir = configBasedir || (options.filepath ? path.dirname(options.filepath) : process.cwd())
```

Comment:
There is no file name with the Node.js API, the block of code is being passed in directly. That block of code can be associated with a file name using the `codeFilename` option.